cdap-proto/src/main/java/co/cask/cdap/proto/SystemServiceLiveInfo.java

```diff
@@ -30,7 +29,7 @@
   private final List<Containers.ContainerInfo> containers;
 
   public SystemServiceLiveInfo(List<Containers.ContainerInfo> containers) {
-    this.containers = ImmutableList.copyOf(containers);
+    this.containers = Collections.unmodifiableList(containers);
```

Comment:
This is not the same. You need to make a copy too.

`Collections.unmodifiableList(new ArrayList<>(containers))`